analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -19,20 +19,265 @@
+
+            val repositoryNode = node["pkgMeta"]["repository"]
+            if ( repositoryNode != null ) {
+                result = repositoryNode["type"].textValueOrEmpty()
+            }
+
+            return result
+        }
+
+        private fun extractRepositoryUrl(node: JsonNode): String {
+            var result = ""
+
+            val repositoryNode = node["pkgMeta"]["repository"]
+            if (repositoryNode != null) {
+                result = repositoryNode["url"].textValueOrEmpty()
+            }
+
+            if (result.isEmpty()) {
+                result = node["pkgMeta"]["_source"].textValueOrEmpty()
+            }
+
+            return result
+        }
+
+        private fun extractRevision(node: JsonNode): String {
+            val resolutionNode = node["pkgMeta"]["_resolution"]
+            if (resolutionNode == null) {
+                return ""
+            }
+
+            var result = resolutionNode["commit"].textValueOrEmpty()
+
+            if (result.isEmpty()) {
+                result = resolutionNode["tag"].textValueOrEmpty()
+            }
+
+            return result
+        }
+
+        private fun extractVcsInfo(node: JsonNode): VcsInfo {
+            return VcsInfo(
+                    type = extractRepositoryType(node),
+                    url = extractRepositoryUrl(node),
+                    revision = extractRevision(node)
+            )
+        }
+
+        private fun extractDeclaredLicenses(node: JsonNode): SortedSet<String> {
+            val result = mutableSetOf<String>()
+
+            val license = node["pkgMeta"]["license"].textValueOrEmpty()
+            if (!license.isEmpty()) {
+                result.add(license)
+            }
+
+            return result.toSortedSet()
+        }
+
+        private fun extractPackage(node: JsonNode ): Package {
+            val vcsInfo = extractVcsInfo(node)
+            return Package(
+                    id = extractPackageId(node),
+                    declaredLicenses = extractDeclaredLicenses(node),
+                    description = node["pkgMeta"]["description"].textValueOrEmpty(),
+                    homepageUrl = node["pkgMeta"]["homepage"].textValueOrEmpty(),
+                    binaryArtifact = RemoteArtifact.EMPTY,
+                    sourceArtifact = RemoteArtifact.EMPTY, // TODO: implement me!
+                    vcs = vcsInfo,
+                    vcsProcessed = vcsInfo.normalize()
+            )
+        }
+
+        private fun extractPackages(node: JsonNode): Map<String, Package> {
+            val result = mutableMapOf<String, Package>()
+
+            val stack = Stack<JsonNode>()
+            stack.push(node)
+
+            while (!stack.empty()) {
+                val currentNode = stack.pop()
+                val pkg = extractPackage(currentNode)
+                result.put("${pkg.id.name}:${pkg.id.version}", pkg)
+
+                val children = currentNode["dependencies"].fields().asSequence().map { it.value }
```

Comment:
I would assume `asSequence()` is not required here.

Reply:
field().map doesn t work